go-controller/pkg/ovn/gateway/gateway.go

```diff
@@ -0,0 +1,98 @@
+package gateway
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/pkg/errors"
+
+	kapi "k8s.io/api/core/v1"
+)
+
+// GetOvnGateways return all created gateways.
+func GetOvnGateways() ([]string, string, error) {
+	out, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=name", "find",
+		"logical_router",
+		"options:chassis!=null")
+	if err != nil {
+		return nil, stderr, err
+	}
+	if out == "" {
```

Comment:
Hmm, this changes the behavior of the code slightly. If we have a `NodePort` type service with a `ClusterIP` defined, then in https://github.com/ovn-org/ovn-kubernetes/blob/9b454e9969bc166a3cc19130a16fc8dd0f55ba85/go-controller/pkg/ovn/service.go#L325

we, today, still perform some subsequent actions if the gateway is empty. We won't do with this patch, now that it will return an error. I am not actually sure it today actually leads to any concrete setup (given that the gateway is an empty string) but I just want to raise that point.

Reply:
Good catch, all the functions return an error on empty but this, so I wanted to keep the API behaviour consistent
I didn't see this err checking
https://github.com/ovn-org/ovn-kubernetes/blob/9b454e9969bc166a3cc19130a16fc8dd0f55ba85/go-controller/pkg/ovn/service.go#L326-L328
the other places that use them check if err = nil and iterate on an empty array that is a noop
I'll change it to not return err and skip